### PR TITLE
Article content edit

### PR DESCRIPTION
**Pull request by austinEng**
This PR attaches an event to the article edit form which pulls content from the Medium editor on submission.  Article content can now be updated from the admin panel

This also fixes dependency problems that were throwing errors and also adds a yarn.lock file which is a newer dependency management tool

### CODE REVIEW

File: src/controllers/admin/articles/index.js [hunk 1]
```diff
@@ -40,17 +41,22 @@ router.post('/:id/edit', function(req, res) {
       content: req.fields.content,
       parent: req.fields.parent,
     },
-
   };
 
+  if (!mongoose.Types.ObjectId.isValid(req.fields.parent)) {
```
Comment (tsuny1) [inline src/controllers/admin/articles/index.js:46]: What do these two functions do?

File: src/react/views/admin/articles/edit.js [hunk 2]
```diff
@@ -60,6 +65,10 @@ class ArticleEdit extends Component {
 }
 
 export default class ArticleEditPage extends Component {
+  getArticleContent() {
+    return this.articleEdit.contentEditor.serialize()['element-0'].value;
```
Comment (tsuny1) [inline src/react/views/admin/articles/edit.js:69]: not sure what this does. What is element-0 in the URL stand for>

File: src/react/views/admin/articles/edit.js [hunk 3]
```diff
@@ -89,7 +98,7 @@ export default class ArticleEditPage extends Component {
                                   <div className="row"><br /></div>
                                   <div className="row">
                                       <ArticleLayout>
-                                          <ArticleEdit article={article} />  
+                                          <ArticleEdit ref={(el) => this.articleEdit = el} article={article} />
```
Comment (tsuny1) [inline src/react/views/admin/articles/edit.js:101]: what is ref field and how is it used?

Comment (austinEng) [reply src/react/views/admin/articles/edit.js:101]: "React supports a special attribute that you can attach to any component. The `ref` attribute takes a callback function, and the callback will be executed immediately after the component is mounted or unmounted."

https://facebook.github.io/react/docs/refs-and-the-dom.html

File: src/react/views/admin/articles/edit.js (hunk 2 above)
Comment (austinEng) [reply src/react/views/admin/articles/edit.js:69]: We're using this module https://github.com/yabwe/medium-editor. 

`.serialize()`: returns a JSON object with elements contents

We only have one element and this is just the naming convention they use

File: src/controllers/admin/articles/index.js (hunk 1 above)
Comment (austinEng) [reply src/controllers/admin/articles/index.js:46]: checks if it's a valid id in the database. There were cases when `parent` was the string "null"

File: src/react/views/admin/articles/edit.js (hunk 2 above)
Comment (tsuny1) [reply src/react/views/admin/articles/edit.js:69]: ok

**Review by tsuny1 — APPROVED** [review]

**Pull request merged**